dashboard/src/components/api/json-rpc/che-json-rpc-master-api.ts

```diff
@@ -36,21 +36,73 @@ const UNSUBSCRIBE: string = 'unsubscribe';
  * @author Ann Shumilova
  */
 export class CheJsonRpcMasterApi {
+  private $log: ng.ILogService;
+  private $timeout: ng.ITimeoutService;
   private cheJsonRpcApi: CheJsonRpcApiClient;
   private clientId: string;
 
-  constructor (client: ICommunicationClient, entrypoint: string) {
+  private maxReconnectionAttempts = 5;
+  private reconnectionAttemptNumber = 0;
+  private reconnectionDelay = 30000;
+
+  constructor (client: ICommunicationClient,
+               entrypoint: string,
+               $log: ng.ILogService,
+               $timeout: ng.ITimeoutService) {
+    this.$log = $log;
+    this.$timeout = $timeout;
+
+    client.addListener('open', () => this.onConnectionOpen());
+    client.addListener('close', () => this.onConnectionClose(entrypoint));
+
     this.cheJsonRpcApi = new CheJsonRpcApiClient(client);
     this.connect(entrypoint);
   }
 
+  onConnectionOpen(): void {
+    this.$log.log('WebSocket connection is opened.');
```

Comment:
should we keep this log ?

Reply:
We could keep this log in case if it is re-connection
```
if (this.reconnectionAttemptNumber !== 0) {
  this.$log.log('...');
}
```
wdyt?

Reply:
yep